ui/src/flux/components/SourceDropdown.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {PureComponent} from 'react'
+
+import Dropdown from 'src/reusable_ui/components/dropdowns/Dropdown'
+
+import {Service, Source, ServiceLinks, SourceLinks} from 'src/types'
+
+interface Props {
+  service: Service
+  services: Service[]
+  sources: Source[]
+  allowInfluxQL: boolean
+  onChangeService: (service: Service, source: Source) => void
+}
+
+interface SourceDropdownItem {
+  sourceID: string
+  serviceID?: string
+  links: ServiceLinks | SourceLinks
+}
+
+class SourceDropdown extends PureComponent<Props> {
+  public render() {
+    return (
+      <Dropdown
```

Comment:
if dropdown items returns an empty array, do you still want to show the dropdown?

Reply:
Same thing here, if there's no sources with Flux services then the user should never see the page anyway